src/pages/content/components/twitter/pageHandlers.tsx

```diff
@@ -0,0 +1,172 @@
+    /* 
+      Initial analysis: determine safe tweet if cache entry for current url not present
+      - first batch will always include the linked tweet (first tweet in batch length of 1)
+      - second batch will include the linked tweet and any surrounding tweets (either replies or the original post) (first tweet in batch length of > 1)
+      - if no second batch (no surrounding tweets) then linked tweet is the original post and has no replies, thus does not need to be analyzed
+    */
+    if (!tweetStatusPageInitialAnalysis.isSafeTweetDetermined) {
+      // save the first tweet in the first batch as the linked tweet (always the first tweet that is loaded on a status page)
+      if (tweetStatusPageInitialAnalysis.batchNumberCounter === 0 && index == 0) {
+        tweetStatusPageInitialAnalysis.linkedTweet = tweet;
+        tweetStatusPageInitialAnalysis.batchNumberCounter++;
+        return;
+      }
+
+      // compare the first tweet of the second batch to the linked tweet to determine if it is a reply or the original post
+      if (tweetStatusPageInitialAnalysis.batchNumberCounter === 1 && index == 0) {
+        // [dont need to do tweet.isEqualNode(tweetStatusPageInitialAnalysis.linkedTweet) here, can just use this first tweet of second batch]
+
+        // save safe tweet info
+        const { tweetHandle, displayName } = getTweetInfo(tweet);
+        tweetSafeInfoMemoryCache[pathname] = { tweetHandle, displayName };
+        tweetStatusPageInitialAnalysis.isSafeTweetDetermined = true;
+      }
+    }
+
+    // once safe tweet is determined, then can proceed to do analysis and ui modifications
+    if (!tweetSafeInfoMemoryCache[pathname]) return;
+    // check if the tweet has already been analyzed
+    if (tweet.getAttribute("data-dl-tweet-check")) return;
+
+    const { tweetHandle, displayName, tweetText, isRepliedTo, isLinkedTweet } = getTweetInfo(tweet);
+
+    // if the tweet handle is safe, then it's not a phishing handle (add to safe handle list)
+    if (tweetHandle.toLowerCase() === tweetSafeInfoMemoryCache[pathname].tweetHandle.toLowerCase()) {
+      handleOpTweet(tweet);
+      return;
+    } else {
+      handleAdTweet(tweet);
+
+      // if the tweet text content consists of only numbers, then it's sus. Add red background the tweet
+      if (/^[0-9]+$/.test(tweetText)) {
+        handleSusTweet(tweet, isLinkedTweet, "onlyNumbers", "BG_RED");
+        return;
+      }
+
+      // only hide addresses if not from the op
+      if (!!tweetText) handleTweetWithAddress(tweet, tweetText, isLinkedTweet);
+    }
+
+    // if the tweet handle is the same as the page handle, then it's sus. Add red background the tweet
+    // [can improve due to false negatives with homoglyphic attacks in the username that cant be detected by equality. maybe use levenshtein distance fuzzy matching on username as well]
+    if (displayName.toLowerCase() == tweetSafeInfoMemoryCache[pathname].displayName.toLowerCase()) {
```

Comment:
exact matching is imo bad, previously we used lev distance for this, so why change it?